client_option.go

```diff
@@ -0,0 +1,63 @@
+package gax
+
+type ClientOption interface {
+	Resolve(*clientSettings)
+}
+
+type clientOptions []ClientOption
+
+func (opts clientOptions) Resolve(s *clientSettings) *clientSettings {
+	for _, opt := range opts {
+		opt.Resolve(s)
+	}
+	return s
+}
+
+type clientSettings struct {
+	apiName    string
+	apiVersion string
+	endpoint   string
+	scopes     []string
```

Comment:
It looks like we have `apiName` and `apiVersion` for our use, then `endpoint` and `scopes` for gRPC's use right?

I wonder if it makes more sense to have
```
struct {
  apiName, apiVersion string
  grpcOpts []grpc.DialOption
}
```
Then in the Dial function, you can just do `grpc.Dial(settings.endpoint, settings.grpcOpts...)`. Of course, we can provide utility function to convert a grpc.DialOpt into ClientOption. This prevents us from needing to do anything if GRPC adds more options in the future. That said, hardcoding these fields make the API a little cleaner. Up to you.